src/main/java/com/management/controller/CarController.java

```diff
@@ -11,60 +13,65 @@
 import java.util.List;
 import java.util.Optional;
 
+import javax.validation.Valid;
+
 @RestController
 @RequestMapping("/cars")
 public class CarController {
 
 	@Autowired
-	private CarRepository carRepository;
+	private CarService carService;
 
 	@GetMapping
-	public List<Car> getAllCars() {
-		return carRepository.findAll();
+	public List<CarModel> getAllCars() {
+	    return carService.findAllCars();
 	}
 
 	@PostMapping
-	public ResponseEntity<?> addCar(@RequestBody Car car) {
-		// Verifica se já existe um carro com a mesma marca, modelo e ano
-		Optional<Car> existingCar = carRepository.findByBrandAndModelAndYear(car.getBrand(), car.getModel(),
-				car.getYear());
-		if (existingCar.isPresent()) {
-			// Se já existir, retorna uma resposta de conflito
-			return ResponseEntity.status(HttpStatus.CONFLICT).body("Carro já incluído anteriormente.");
-		}
-		// Se não existir, salva o novo carro
-		Car savedCar = carRepository.save(car);
-		return ResponseEntity.ok(savedCar);
+	public ResponseEntity<?> addCar(@RequestBody @Valid CarModel carModel) {
+	    try {
+	        Optional<CarModel> savedCar = carService.saveCar(carModel);
+	        if (savedCar.isPresent()) {
+	            return ResponseEntity.status(HttpStatus.CREATED).body(savedCar.get());
+	        } else {
+	            return ResponseEntity.status(HttpStatus.CONFLICT).body("Carro já incluído anteriormente.");
+	        }
+	    } catch (IllegalArgumentException e) {
+	        // Aqui você trata a exceção específica lançada pelo seu serviço
+	        return ResponseEntity.badRequest().body(e.getMessage());
+	    }
 	}
 
+
 	@GetMapping("/{id}")
-	public ResponseEntity<Car> getCarById(@PathVariable Long id) {
-		return carRepository.findById(id).map(car -> ResponseEntity.ok().body(car))
-				.orElse(ResponseEntity.notFound().build());
+	public ResponseEntity<CarModel> getCarById(@PathVariable Long id) {
+	    return carService.findCarById(id)
+	            .map(car -> ResponseEntity.ok().body(car))
+	            .orElse(ResponseEntity.notFound().build());
 	}
-
+	
 	@DeleteMapping("/{id}")
 	public ResponseEntity<String> deleteCar(@PathVariable Long id) {
-		// Verifica se o carro com o ID fornecido existe
-		boolean exists = carRepository.existsById(id);
-		if (exists) {
-			carRepository.deleteById(id);
-			return ResponseEntity.ok("Carro deletado com sucesso!");
-		} else {
-			// return ResponseEntity.notFound().build();
-			return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Carro não encontrado!");
-		}
+	    try {
+	        carService.deleteCar(id);
+	        return ResponseEntity.ok("Carro deletado com sucesso!");
```

Comment:
Não existe um código http mais apropriado para deleção? ao invés de ok?

Reply:
yes, there is. The code is 204 - NO CONTENT